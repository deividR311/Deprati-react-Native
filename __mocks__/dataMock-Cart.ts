export const DataMock_Cart = {
  carts: [
    {
      appliedOrderPromotions: [],
      appliedProductPromotions: [],
      appliedVouchers: [],
      calculated: true,
      code: '4809002',
      deliveryAddress: {
        cellphoneNumber: '34949494',
        cellphonePreffix: '09',
        city: 'COT-287',
        country: {
          isocode: 'EC',
          name: 'Ecuador'
        },
        defaultAddress: true,
        email: 'alexis.enriquezca@gmail.com',
        firstName: 'Alejandra ',
        formattedAddress: 'Calle 6, Casa 23, Cotopaxi, Alaquez',
        id: '8819555172375',
        idNumber: '1721480299',
        idType: 'CEDULA',
        line1: 'Calle 6',
        line2: 'Casa 23',
        otherInfo: 'Laura',
        phone: '',
        region: {
          countryIso: 'EC',
          isocode: 'EC-COT',
          isocodeShort: '193',
          name: 'Cotopaxi'
        },
        shippingAddress: true,
        town: 'Alaquez',
        visibleInAddressBook: true
      },
      deliveryItemsQuantity: 1,
      deliveryOrderGroups: [
        {
          entries: [
            {
              basePrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              basePriceWithTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,60',
                priceType: 'BUY',
                value: 2.5984
              },
              basePriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              deliveryMode: {
                code: 'standard'
              },
              deliveryTimeRange: '5 a 7 días laborables',
              entryNumber: 0,
              giftPacking: false,
              pickup: false,
              product: {
                acceptGiftPacking: true,
                availableForPickup: false,
                baseOptions: [
                  {
                    selected: {
                      code: '226010190802030008',
                      priceData: {
                        currencyIso: 'USD',
                        formattedValue: '$2,32',
                        priceType: 'BUY',
                        value: 2.32
                      },
                      stock: {
                        stockLevel: 11,
                        stockLevelStatus: 'inStock'
                      },
                      url: '/camiseta-estampada/p/226010190802030008',
                      variantOptionQualifiers: []
                    },
                    variantType: 'GenericVariantProduct'
                  }
                ],
                baseProduct: '000226010190802030',
                categories: [
                  {
                    code: 'T8023',
                    name: '10',
                    parentCategoryName: 'Talla',
                    url: '/10/c/T8023'
                  },
                  {
                    code: 'C0110601',
                    name: 'Blanco',
                    parentCategoryName: 'Color',
                    url: '/blanco/c/C0110601'
                  },
                  {
                    code: '0701',
                    name: 'MUJERES',
                    url: '/mujeres/c/0701'
                  }
                ],
                code: '226010190802030008',
                deliveryModes: [
                  {
                    code: 'standard'
                  },
                  {
                    code: 'pickup'
                  }
                ],
                ean: '15282579',
                images: [
                  {
                    format: 'zoom',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
                  },
                  {
                    format: 'product',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
                  },
                  {
                    format: 'thumbnail',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
                  },
                  {
                    format: 'cartIcon',
                    imageType: 'PRIMARY',
                    url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
                  }
                ],
                isGiftProduct: false,
                isPromotionSpecialPrice: false,
                ivaCondition: '12%',
                name: 'CAMISETA ESTAMPADA',
                purchasable: true,
                stock: {
                  stockLevel: 11,
                  stockLevelStatus: 'inStock'
                },
                url: '/camiseta-estampada/p/226010190802030008'
              },
              quantity: 1,
              subtotal: {
                currencyIso: 'USD',
                formattedValue: '$23,20',
                priceType: 'BUY',
                value: 23.2
              },
              totalDiscount: {
                currencyIso: 'USD',
                formattedValue: '$20,88',
                priceType: 'BUY',
                value: 20.88
              },
              totalPrice: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              totalPriceWithoutTaxes: {
                currencyIso: 'USD',
                formattedValue: '$2,32',
                priceType: 'BUY',
                value: 2.32
              },
              updateable: true,
              volume: 0,
              weight: 0
            }
          ],
          totalPriceWithTax: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5999999999999996
          }
        }
      ],
      entries: [
        {
          basePrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          basePriceWithTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,60',
            priceType: 'BUY',
            value: 2.5984
          },
          basePriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          deliveryMode: {
            code: 'standard'
          },
          deliveryTimeRange: '5 a 7 días laborables',
          entryNumber: 0,
          giftPacking: false,
          pickup: false,
          product: {
            acceptGiftPacking: true,
            availableForPickup: false,
            baseOptions: [
              {
                selected: {
                  code: '226010190802030008',
                  priceData: {
                    currencyIso: 'USD',
                    formattedValue: '$2,32',
                    priceType: 'BUY',
                    value: 2.32
                  },
                  stock: {
                    stockLevel: 11,
                    stockLevelStatus: 'inStock'
                  },
                  url: '/camiseta-estampada/p/226010190802030008',
                  variantOptionQualifiers: []
                },
                variantType: 'GenericVariantProduct'
              }
            ],
            baseProduct: '000226010190802030',
            categories: [
              {
                code: 'T8023',
                name: '10',
                parentCategoryName: 'Talla',
                url: '/10/c/T8023'
              },
              {
                code: 'C0110601',
                name: 'Blanco',
                parentCategoryName: 'Color',
                url: '/blanco/c/C0110601'
              },
              {
                code: '0701',
                name: 'MUJERES',
                url: '/mujeres/c/0701'
              }
            ],
            code: '226010190802030008',
            deliveryModes: [
              {
                code: 'standard'
              },
              {
                code: 'pickup'
              }
            ],
            ean: '15282579',
            images: [
              {
                format: 'zoom',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h67/h07/8890805157918/15282579-0_product_515Wx772H'
              },
              {
                format: 'product',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h26/h0b/8890805288990/15282579-0_product_300Wx450H'
              },
              {
                format: 'thumbnail',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h38/h0e/8890805354526/15282579-0_product_96Wx144H'
              },
              {
                format: 'cartIcon',
                imageType: 'PRIMARY',
                url: 'https://imagestg.deprati.com.ec/sys-master/images/h49/h11/8890805420062/15282579-0_product_65Wx97H'
              }
            ],
            isGiftProduct: false,
            isPromotionSpecialPrice: false,
            ivaCondition: '12%',
            name: 'CAMISETA ESTAMPADA',
            purchasable: true,
            stock: {
              stockLevel: 11,
              stockLevelStatus: 'inStock'
            },
            url: '/camiseta-estampada/p/226010190802030008'
          },
          quantity: 1,
          subtotal: {
            currencyIso: 'USD',
            formattedValue: '$23,20',
            priceType: 'BUY',
            value: 23.2
          },
          totalDiscount: {
            currencyIso: 'USD',
            formattedValue: '$20,88',
            priceType: 'BUY',
            value: 20.88
          },
          totalPrice: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          totalPriceWithoutTaxes: {
            currencyIso: 'USD',
            formattedValue: '$2,32',
            priceType: 'BUY',
            value: 2.32
          },
          updateable: true,
          volume: 0,
          weight: 0
        }
      ],
      guid: '15901dbd-9966-4239-9a4d-fecc4f747589',
      net: true,
      orderDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      pickupItemsQuantity: 0,
      pickupOrderGroups: [],
      productDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      sellWithTaxes: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      sellWithoutTaxWithoutInterest: {
        currencyIso: 'USD',
        formattedValue: '$2,31',
        priceType: 'BUY',
        value: 2.31
      },
      sellWithoutTaxes: {
        currencyIso: 'USD',
        formattedValue: '$0,00',
        priceType: 'BUY',
        value: 0
      },
      site: 'deprati',
      store: 'deprati',
      subTotal: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalDiscounts: {
        currencyIso: 'USD',
        formattedValue: '$20,88',
        priceType: 'BUY',
        value: 20.88
      },
      totalItems: 1,
      totalPrice: {
        currencyIso: 'USD',
        formattedValue: '$2,32',
        priceType: 'BUY',
        value: 2.32
      },
      totalPriceWithTax: {
        currencyIso: 'USD',
        formattedValue: '$2,60',
        priceType: 'BUY',
        value: 2.6
      },
      totalTax: {
        currencyIso: 'USD',
        formattedValue: '$0,28',
        priceType: 'BUY',
        value: 0.28
      },
      totalVolume: 0,
      totalWeight: 0.25,
      user: {
        name: 'Alexis pruebas ENRIQUEZ',
        uid: 'alexis.enriquezca@gmail.com'
      },
      potentialOrderPromotions: [
        {
          consumedEntries: [],
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_A',
            description: 'Version 2 Hybris - PRUEBA APP',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        },
        {
          consumedEntries: [],
          description:
            'Completa $150 y  obten envío gratis. Aplica restricciones.',
          promotion: {
            code: '10012022_EGCE_PRUEBAAPP1_C',
            description: 'Version 2 Hybris',
            endDate: '2024-01-01T04:59:59+0000',
            promotionType: 'Promoción basada en regla'
          }
        }
      ],
      potentialProductPromotions: [],
      totalUnitCount: 1
    }
  ]
};
