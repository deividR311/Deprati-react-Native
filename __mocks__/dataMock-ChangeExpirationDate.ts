export const DataMock_ChangeExpirationDate = {
  TEXT_NEWDATE: 'Selecciona el día',
  infoCreditBalance: {
    clientName: 'PATRICIA L. DE UGALDE',
    cardNumber: '1xxxxx6',
    affiliateDate: '2000',
    totalAmount: '5.500,00',
    amountSpent: '724,35',
    aditional: '00',
    availableAmount: '4.775,65',
    pastDueDays: 0,
    nextPayDay: 'Abril 14 de 2023',
    nextAmountToPay: '105,08',
    numeroTarjetaAdicionalDisplay: '14XXX86 00',
    clientFirstName: 'Prueba'
  },
  dataCurrentDate: {
    cuttingDay: 14,
    dueDate: '20230514',
    day: '14'
  },
  dataChangeDate: null,
  selectedDay: '2',
  isAcceptTerms: false,
  showWarning: false,
  isLoadingChangeDate: false,
  isLoadingCurrentDate: false,
  showSucces: false,
  showErrorSucces: false,
  expirationDateContent: {
    legend:
      'No puedes cambiar tu fecha máxima de pago si tienes una deuda pendiente y <b>este cambio se hará efectivo a partir del próximo mes.</b>',
    popup_title:
      '¿Estás seguro de cambiar la fecha máxima de pago de tu crédito De Prati?',
    popup_paragraphs: [
      'Ten en cuenta:',
      '1. Esta acción no se puede deshacer y el cambio se hará efectivo a partir de tu próximo mes.',
      '2. Si haces el cambio de fecha <b>deberás pagar la cuota del mes en curso</b> y este cambio se hará efectivo a partir del próximo mes.'
    ],
    popup_accept_action: 'SI, ESTOY DE ACUERDO',
    popup_cancel_action: 'NO, CANCELAR'
  },
  selectRef: {
    current: {
      reset: jest.fn(),
      openDropdown: jest.fn(),
      closeDropdown: jest.fn()
    }
  },
  setSelectedDay: jest.fn(),
  setIsAcceptTerms: jest.fn(),
  handleChangeDate: jest.fn(),
  setShowWarning: jest.fn(),
  handleCloseSuccess: jest.fn(),
  handleCloseError: jest.fn(),
  handleCloseWarning: jest.fn()
};
