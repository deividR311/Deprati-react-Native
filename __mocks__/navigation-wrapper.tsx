import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { COLORS } from '../src/application/common';
import { ProvideWrapper } from './provider-wrapper';

jest.useFakeTimers();
jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

const Stack = createNativeStackNavigator();

export const NavigationWrapper = ({ children, screen, initialParams }: any) => (
  <ProvideWrapper>
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="default"
        screenOptions={{
          headerShown: true,
          headerShadowVisible: false,
          headerTitleAlign: 'left',
          contentStyle: { backgroundColor: COLORS.WHITE }
        }}>
        {screen && (
          <Stack.Screen
            name="default"
            initialParams={initialParams ?? {}}
            component={screen}
          />
        )}
        {children && <Stack.Screen name="default" component={() => children} />}
      </Stack.Navigator>
    </NavigationContainer>
  </ProvideWrapper>
);
