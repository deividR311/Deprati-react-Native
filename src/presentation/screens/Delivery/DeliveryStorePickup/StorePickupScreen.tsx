//Libs
import React from 'react';
import { FlatList, ScrollView, Text, View } from 'react-native';
import { List } from 'react-native-paper';
//hooks
import useStorePickup, { CITY, STORE } from './hook/useStorePickup.hook';

//components
import InfoStoreComponent from './components/InfoStoreComponent';
import InfoPersonalComponent from './components/InfoPersonalComponent';
import ItemProduct from './components/ItemProduct';
import { StorePickupSkeleton } from './components/StorePickupSkeleton/StorePickupSkeleton';
import TemplatePage from '../../../common-components/template-page';
import SelectInput from '../../../common-components/inputs/SelectInput';
import { ButtonOutlined } from '../../../common-components/buttons/Button';
import {
  RightIconAccordion,
  stylesAccordion
} from '../../../common-components/RightIconAccordion/RightIconAccordion';
import { styles } from './StorePickup.StyleSheet';

export default function StorePickupScreen() {
  const {
    dataCart,
    //formik
    values,
    errorsFormik,
    onChange,
    onChangeText,
    isLoading,
    //ref
    storeRef,
    cityRef,
    handleReset,
    /////
    cities,
    stores,
    currentStore,
    ////
    accordion
  } = useStorePickup();

  return (
    <TemplatePage
      loading={isLoading}
      skeleton={<StorePickupSkeleton />}
      error={false}>
      <ScrollView contentContainerStyle={styles.container}>
        <View style={styles.content}>
          <View style={styles.contentTitle}>
            <Text style={styles.title}>
              Elige la tienda más cercana para ti.
            </Text>
          </View>
          <View style={{}}>
            <SelectInput
              testID={'deliverythirdparty_city'}
              selectRef={cityRef}
              styles={styles.inputCity}
              error={Object.keys(errorsFormik).includes(CITY)}
              items={cities}
              onChange={(text: string) => onChange(CITY, text)}
              value={values.city}
              inputBasic={true}
              label="Selecciona ciudad"
            />
            <SelectInput
              testID={'deliverythirdparty_store'}
              selectRef={storeRef}
              styles={styles.inputStore}
              error={Object.keys(errorsFormik).includes(STORE)}
              items={stores}
              onChange={(text: string) => onChange(STORE, text)}
              value={values.store}
              inputBasic={true}
              label="Selecciona la tienda"
            />
            {!currentStore && (
              <ButtonOutlined
                title="LIMPIAR"
                style={{
                  ...styles.buttonClean,
                  width: 180,
                  marginVertical: 14
                }}
                onPress={handleReset}
              />
            )}
            {currentStore && (
              <>
                <InfoStoreComponent store={currentStore} />
                <View style={styles.contentButton}>
                  <ButtonOutlined
                    title="LIMPIAR"
                    style={styles.buttonClean}
                    onPress={handleReset}
                  />
                </View>
                <InfoPersonalComponent
                  values={values}
                  onChangeText={onChangeText}
                  errors={errorsFormik}
                />
              </>
            )}
          </View>
        </View>
        <List.Accordion
          style={[styles.accordion, stylesAccordion.container__accordion]}
          titleStyle={styles.accordion_title}
          expanded={accordion.isExpanded}
          onPress={() => accordion.setIsExpanded(!accordion.isExpanded)}
          title={'Artículos para retiro en tienda'}
          right={RightIconAccordion}>
          <FlatList
            showsHorizontalScrollIndicator={false}
            horizontal
            keyExtractor={(item, index) => index.toString()}
            data={dataCart?.entries ?? []}
            renderItem={x => {
              return <ItemProduct data={x.item} />;
            }}
          />
        </List.Accordion>
      </ScrollView>
    </TemplatePage>
  );
}
