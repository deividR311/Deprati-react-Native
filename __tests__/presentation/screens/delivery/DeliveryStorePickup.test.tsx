import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationWrapper } from '../../../../__mocks__/navigation-wrapper';
import StorePickupScreen from '../../../../src/presentation/screens/Delivery/DeliveryStorePickup/StorePickupScreen';

const hookApi =
  '../../../../src/infrastucture/apis/checkout/pickupStore/pointOfService.api';
jest.mock(hookApi, () => ({
  useLazyPickupStoreRequest: () => [
    jest.fn(),
    {
      data: {
        posGroups: []
      },
      isLoading: true,
      isSuccess: false,
      isError: false
    }
  ],
  useSelectPOSRequest: () => [
    jest.fn(),
    {
      data: {
        data: {}
      },
      isLoading: false,
      isSuccess: false,
      isError: false
    }
  ]
}));

describe('DeliveryStorePickup', () => {
  const mockInitParams = {
    dataCart: {},
    setTitleContinueButton: jest.fn(),
    enableContinueButton: jest.fn(),
    showActivityIndicatorContinueButton: jest.fn(),
    showTotalsBand: jest.fn(),
    setCurrentStep: jest.fn()
  };

  it('renders correctly in Snapshot Loaging', () => {
    const screenTest = render(
      <NavigationWrapper
        screen={StorePickupScreen}
        initialParams={mockInitParams}
      />
    );
    expect(screenTest?.toJSON()).toMatchSnapshot();
  });

  /*  it('renders correctly in Snapshot', () => {
    jest
      .spyOn(require(hookApi), 'useLazyPickupStoreRequest')
      .mockReturnValueOnce([
        jest.fn(),
        {
          isLoading: false,
          isSuccess: true,
          isError: false,
          data: {
            posGroups: [{ code: 'cali', name: 'cali', postList: [] }],
          },
        },
      ])
    const screenTest = render(
      <NavigationWrapper
        screen={StorePickupScreen}
        initialParams={mockInitParams}
      />,
    )
    expect(screenTest?.toJSON()).toMatchSnapshot()
  }) */
});
