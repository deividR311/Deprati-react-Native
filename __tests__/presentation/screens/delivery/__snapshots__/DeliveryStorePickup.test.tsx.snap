// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryStorePickup renders correctly in Snapshot Loaging 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        Object {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        hideShadow={true}
        largeTitleHideShadow={false}
        title="default"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 20,
                "width": "100%",
              }
            }
          >
            <View
              onLayout={[Function]}
              style={Object {}}
            >
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 25,
                    "key": "title",
                    "marginLeft": 7,
                    "marginTop": 20,
                    "overflow": "hidden",
                    "width": 220,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 40,
                    "key": "dropDown1",
                    "marginLeft": 7,
                    "marginTop": 30,
                    "overflow": "hidden",
                    "width": 300,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 40,
                    "key": "dropDown2",
                    "marginLeft": 7,
                    "marginTop": 10,
                    "overflow": "hidden",
                    "width": 300,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 120,
                    "key": "address",
                    "marginLeft": 7,
                    "marginTop": 20,
                    "overflow": "hidden",
                    "width": 300,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignSelf": "auto",
                    "backgroundColor": "#E1E9EE",
                    "borderRadius": 4,
                    "height": 40,
                    "key": "btn-clean",
                    "marginLeft": 35,
                    "marginTop": 20,
                    "overflow": "hidden",
                    "width": 120,
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": "100%",
                        "position": "absolute",
                        "width": "100%",
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "translateX": undefined,
                          },
                        ],
                      },
                    ]
                  }
                >
                  <BVLinearGradient
                    colors={
                      Array [
                        4292995566,
                        4294113532,
                        4292995566,
                      ]
                    }
                    endPoint={
                      Object {
                        "x": 1,
                        "y": 0,
                      }
                    }
                    locations={null}
                    startPoint={
                      Object {
                        "x": 0,
                        "y": 0,
                      }
                    }
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <Modal
            hardwareAccelerated={false}
            transparent={true}
            visible={true}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffffb3",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                accessibilityRole="progressbar"
                accessibilityState={
                  Object {
                    "busy": true,
                  }
                }
                accessible={true}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "height": 48,
                      "opacity": 1,
                      "width": 48,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 48,
                          "transform": Array [
                            Object {
                              "rotate": "45deg",
                            },
                          ],
                          "width": 48,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 24,
                            "overflow": "hidden",
                            "width": 48,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 48,
                              "transform": Array [
                                Object {
                                  "translateY": 0,
                                },
                                Object {
                                  "rotate": "-165deg",
                                },
                              ],
                              "width": 48,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "height": 24,
                                "overflow": "hidden",
                                "width": 48,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "borderColor": "#E31A2C",
                                  "borderRadius": 24,
                                  "borderWidth": 4.8,
                                  "height": 48,
                                  "width": 48,
                                }
                              }
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "alignItems": "center",
                        "bottom": 0,
                        "justifyContent": "center",
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "height": 48,
                          "transform": Array [
                            Object {
                              "rotate": "45deg",
                            },
                          ],
                          "width": 48,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "height": 24,
                            "overflow": "hidden",
                            "top": 24,
                            "width": 48,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "height": 48,
                              "transform": Array [
                                Object {
                                  "translateY": -24,
                                },
                                Object {
                                  "rotate": "345deg",
                                },
                              ],
                              "width": 48,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "height": 24,
                                "overflow": "hidden",
                                "width": 48,
                              }
                            }
                          >
                            <View
                              collapsable={false}
                              style={
                                Object {
                                  "borderColor": "#E31A2C",
                                  "borderRadius": 24,
                                  "borderWidth": 4.8,
                                  "height": 48,
                                  "width": 48,
                                }
                              }
                            />
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
